lib/vmdb/config.rb

```diff
@@ -7,19 +7,26 @@ class Config
 
     require_relative 'configuration_encoder'
 
-    def self.clone_auth_for_log(auth)
-      log_auth = auth.deep_clone
-      [:bind_pwd, :amazon_secret].each do |key|
-        log_auth[key] = '********' if log_auth.key?(key)
-        if log_auth.key?(:user_proxies)
-          log_auth[:user_proxies].each do |p|
-            p[key] = '********' if p.key?(key)
-          end
-        end
+
+    PASSWORD_KEYS = [:bind_pwd, :amazon_secret]
+
+    def self.deep_clean_auth_for_log(log_auth)
+      log_auth.each do |key, val|
+        log_auth[key] = deep_clean_auth_for_log(val) if val.kind_of?(Hash)
+
+        log_auth[key] = "********" if PASSWORD_KEYS.include? key
+
+        log_auth[key].each do |p|
+          p.each { |key2, _val2| p[key2] = "********" if PASSWORD_KEYS.include? key2 }
+        end if [:user_proxies].include? key
       end
       log_auth
```

Comment:
`log_auth.each` returns itself, which is being modified. So you can get rid of this line.
But you might leave it for clarity :)